packages/backend/src/payment-method/ilp/connector/core/middleware/telemetry.ts

```diff
@@ -0,0 +1,22 @@
+import { collectTelemetryAmount } from '../../../../../telemetry/transaction-amount'
+import { ILPContext, ILPMiddleware } from '../rafiki'
+
+export function createTelemetryMiddleware(): ILPMiddleware {
+  return async (
+    { request, services, accounts, state }: ILPContext,
+    next: () => Promise<void>
+  ): Promise<void> => {
+    if (
+      services.telemetry &&
+      Number(request.prepare.amount) &&
+      !state.unfulfillable
+    ) {
+      const { code, scale } = accounts.outgoing.asset
+      collectTelemetryAmount(services.telemetry, services.logger, {
+        amount: BigInt(request.prepare.amount),
+        asset: { code: code, scale: scale }
+      })
+    }
+    await next()
```

Comment:
this will end up counting the amount even if the packet is rejected at the destination.

Maybe we can call `next()` first so that this effectively ends up being the last middleware to run (we can also just put it as the last one in the middleware chain), and check if `ctx.response.fulfill` is defined. This way, only fulfilled packets will be tracked.

Reply:
I would go for the 2nd option you mentioned.

Reply:
ended up going for the 1st 😁